src/features/device-state/routes/state-get-v2.ts

```diff
@@ -93,10 +97,10 @@ function buildAppFromRelease(
 		varListInsert(application.application_environment_variable, environment);
 		varListInsert(svc.service_environment_variable, environment);
 		varListInsert(device.device_environment_variable, environment);
-		const dsev = device.device_service_environment_variable.filter(
-			({ service }) => service.__id === svc.id,
-		);
-		varListInsert(dsev, environment);
+		const dsevs = dsevsById[svc.id];
```

Comment:
@thgreasi You can see I actually mentioned this in my original PR: https://github.com/balena-io/open-balena-api/pull/1862#discussion_r1858615955

But wanted to avoid premature optimising it